src/api/responses.rs

```diff
@@ -0,0 +1,220 @@
+// Copyright 2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use std::ops::Deref;
+
+use axum::response::IntoResponse;
+use derive_more::From;
+use serde::{
+    Deserialize,
+    Serialize,
+};
+use serde_json::Value;
+
+use super::extractors::{
+    MessagesQuery,
+    OutputsQuery,
+};
+use crate::types::LedgerInclusionState;
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[serde(untagged)]
+pub(crate) enum ListenerResponse {
```

Comment:
Do we need all these responses in the same `enum`, or could we just use `impl IntoResponse` where appropriate?

Reply:
There are pros and cons to both approaches, but either work.